framework/resource/retryresource/error.go

```diff
@@ -4,6 +4,13 @@ import (
 	"github.com/giantswarm/microerror"
 )
 
+var incompatibleUnderlyingResourceError = microerror.New("incompatible underlying resource")
+
+// isIncompatibleUnderlyingResource asserts incompatibleUnderlyingResourceError.
+func isIncompatibleUnderlyingResource(err error) bool {
+	return microerror.Cause(err) == incompatibleUnderlyingResourceError
+}
```

Comment:
This is private matcher. I wouldn't like to expose that to the user. This gives us freedom to change it w/o consequences.